### ISSUE TITLE
need a function to copy elements in list if the case for any variable

### ISSUE THREAD

**Issue by og76** [JacquesLucke/animation_nodes#483]
in conclusion to the experiments in #474 and #475 (fill list and repeat list) 
in order to clean up and draw a conclusion, **to be easier to follow**

We need an extra _"get **copy expression from socket type** and apply **to any variable**"_ function.
Like

``` py
    # It can use fromSocket or idName
    def getCopyString(self, fromSocket, variableString):
        if not fromSocket.isCopyable: return variableString
        return fromSocket.getCopyExpression().replace("value", variableString)
```
would be used as
``` py
        fromSocket = self.inputs["List"]
        yield "    reList = list(reversed(inList))"

        yield "    for i in range(amount):"
        yield "        if (i % 2) == 0: outList += {}".format(self.getCopyString(fromSocket, "inList"))
        yield "        else: outList += {}".format(self.getCopyString(fromSocket, "reList"))

```
As you can see, the example does copy later in the code, not at level of input variable.
.

For this, **I think we simply need this function in sockets/info.py** or respective nodes. 

No need another "structural" intervention for this, as the copy here is not at the level of sockets, but later in the code, for specific variable.
.

I would not tie to execution/ code_generator (that contains a getCopyExpression and getCopyLine)
That remains as is, dealing only with sockets and final code format, while this is a convenience for use where the case. 

-

ps
_just to restate the initial problem / case for those interested_

both fill list (input element) and repeat list (input list) have the issue of needing to copy the elements, or elements in the list for each instance of repetition

list types are changed (automated) depending on input, so I don't know what element I get and what kind of copy it needs, making a general execution code for all data types

Important: I need to use this later in the execution code, at any variable, and not at the level of input

- if the case, I see in `socket.isCopyable` prop  (or socket class)
- for those isCopyable, copy expression / string is in the sockets files as `getCopyExpression()`

**Comment by og76** [comment]
ATM both problematic nodes mentioned contain the above function and just work
I should probably.
- move the function in sockets/info
- make a clean new pull (or change existing, if they are not hard to follow already)

**Comment by JacquesLucke** [comment]
I actually like your conclusion and I think I will make it like you proposed here + some restrucuring in the sockets/info.py file.
(I want that this file can imports all socket classes first and caches the results of all the functions that are in this file, not sure how to explain this better now, but you'll see)

**Comment by og76** [comment]
sound good
